### ISSUE TITLE
Question related to delayed connection close and TCP CONNECT

### ISSUE THREAD

**Issue by tehnerd** [envoyproxy/envoy#24083]
in https://github.com/envoyproxy/envoy/pull/21423 that delay option was removed for proxy working in CONNECT mode.
Question is: were there any ongoing issues related to this? because in our environment we had an issue w/ npm client connecting through proxy. The issue was that npm registry were clossing connections w/ TCP RST time to time and envoy was sending reset downstream to the npm client. 
Client, which still had some data in tcp buffer, after receiving reset, dropped that data. and from application point of view it was looked like that some amount of bytes in the end of the response were missing.

To mitigate this we configured delayed_close_seconds

However after updating envoy (w/ PR from above include) NPM client starts to see that issue again and we were forced to set envoy.reloadable_features.no_delay_close_for_upgrades to false

Was wonder are there any other way to mitigate this? (assuming that "ignore delay in case of tcp connect would move forward and relodable feature would be removed")

cc @alyssawilk

**Comment by tehnerd** [comment]
That would be awesome. Thanks

**Comment by fuhry** [comment]
To add some further context to this - we found that this behavior is pretty easy to reproduce with Node.js clients using Envoy as a CONNECT proxy. Running `npm ci` in a medium-to-large project checkout, with the `HTTP_PROXY` and `HTTPS_PROXY` env vars set to an Envoy plaintext listener configured with the `dynamic_forward_proxy` http filter, is sufficient to reproduce the behavior 95-100% of the time.

We're not sure why Node.js is the only client to be affected - we spent many months validating Envoy before fully shutting down our previous outbound proxy solution - but making the delayed_close_timeout longer (5 seconds was sufficient in our case) solved the issue.
